server/message_hooks.go

```diff
@@ -34,11 +34,7 @@ func (p *Plugin) MessageWillBePosted(_ *plugin.Context, post *model.Post) (*mode
 					return post, ""
 				}
 
-				if user.RemoteId == nil {
-					continue
-				}
-
-				if isMSTeamsUser(*user.RemoteId, user.Username) {
+				if user.RemoteId != nil && strings.Contains(user.Username, "msteams") {
```

Comment:
This is slightly imprecise, I would use `strings.HasPrefix` instead.

Reply:
IMO you should be looking for `msteams:` in case an account gets created with `msteams` in the name.  Users cannot create usernames with colons so it is safe to assume `msteams:` won't appear.  Also, it was previously decided by PM that username would appear before remote name.

Reply:
cc @jespino